src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1630,6 +1654,72 @@ private void MakeSeparatorList(ReadOnlySpan<char> separators, ref ValueListBuild
             }
         }
 
+        private void MakeSeparatorListVectorized(ref ValueListBuilder<int> sepListBuilder, char c, char? c2 = null, char? c3 = null)
+        {
+            // Constant that defines indices of characters within an AVX-Register
+            const ulong indicesConstant = 0xFEDCBA9876543210;
+            ReadOnlySpan<byte> shuffleConstantData = new byte[] {
```

Comment:
stackalloc this array?

Reply:
Does not actually allocate and seems to reside in assembly data region https://github.com/dotnet/runtime/pull/38001#discussion_r441416404

Reply:
I was kind of hoping for a more explicit way to do this, as this is not clear that this happens behind the scenes. But generates a lot less code, as per the Sharplab link at that comment above.

Reply:
It's actually better to just use `Vector256.Create`. .NET 5 will create a constant for this now at JIT time

Reply:
@tannergooding I must have missed that, and out of interest: can you point me to that change please?

And does this mean the JIT creates a constant, which is then read from memory? So basically the same as with the ROS and reading from there, just with simpler and more intuitive code.

Reply:
@gfoidl: https://github.com/dotnet/runtime/pull/35857 and https://github.com/dotnet/runtime/pull/36267

It creates a method local constant that is properly aligned (unless compiling for small code). In the future it will likely be expanded to support things that are mostly constant as well, although that won't be in for .NET 5.

Reply:
Thanks!

Reply:
Reverted to `Vector.Create` in https://github.com/dotnet/runtime/pull/38001/commits/65cba0ca5ccc2bb55fa8044cb7326e2e0b6a455a